wiring/inc/spark_wiring_wifi.h

```diff
@@ -201,7 +201,7 @@ class WiFiClass : public NetworkClass
     IPAddress resolve(const char* name)
     {
         HAL_IPAddress ip;
-        return (inet_gethostbyname(name, strlen(name), &ip, *this, NULL)<0) ?
+        return (inet_gethostbyname(name, strlen(name), &ip, *this, NULL) != 0) ?
```

Comment:
can you please explain this? I thought the default comparison is not equal to 0?

Reply:
Electron's `inet_gethostbyname` returns a positive value in case of an error, which fails the `<0` check. It sounds also reasonable to make sure that the return value is negative in all the HALs, but the check for `!= 0` sounds good as well, as that mimics the behavior of `gethostbyname_r`.

Reply:
oh, I didn't see the <0 when I originally looked (text wrapping.)  If we change it to != 0 then we should check no other HALs are using >0 codes for success. (I have a feeling the CC3000 does this...) so it might be better to make the Electron HAL conform to the established <0 convention.

Reply:
I don't think that's the case on the Core either: see https://github.com/spark/firmware/blob/develop/hal/src/core/inet_hal.c#L39